src/constraints/storage_constraints.jl

```diff
@@ -74,12 +73,21 @@ end
 
 function add_elec_storage_dispatch_constraints(m, p, b; _n="")
 				
-	# Constraint (4g)-1: state-of-charge for electrical storage - with grid
+    # Constraint (4d): Storage export must be less than total storage dispatch
+    if !isempty(p.export_bins_by_storage[b]) # TODO: Is this if statement needed? 
+        @constraint(m, [b in p.s.storage.attr[b], ts in p.time_steps_with_grid],
+            sum(m[Symbol("dvStorageToGrid"*_n)][b, u, ts] for u in p.export_bins_by_storage[b])
+            <= 
+            m[Symbol("dvDischargeFromStorage"*_n)][b, ts] # TODO: should this be divided by p.s.storage.attr[b].discharge_efficiency ?
```

Comment:
no discharge_efficiency here, that is applied in the SOC constraints (4g)

Reply:
To be sure I understand, is `dvProductionToStorage` on the AC side of the converter (so, before efficiency losses) and `dvDischargeFromStorage` is on the AC side of the inverter (so, after efficiency losses)? So everything is AC